iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/distribute/TableDistributedPlanGenerator.java

```diff
@@ -524,6 +527,120 @@ public List<PlanNode> visitDeviceTableScan(DeviceTableScanNode node, PlanContext
     return resultTableScanNodeList;
   }
 
+  @Override
+  public List<PlanNode> visitTreeDeviceViewScan(TreeDeviceViewScanNode node, PlanContext context) {
+    // left of pair if TreeAlignedDeviceViewScanNode, right is TreeNonAlignedDeviceViewScanNode
+    Map<TRegionReplicaSet, Pair<TreeDeviceViewScanNode, TreeDeviceViewScanNode>> tableScanNodeMap =
+        new HashMap<>();
+
+    for (DeviceEntry deviceEntry : node.getDeviceEntries()) {
+      List<TRegionReplicaSet> regionReplicaSets =
+          analysis.getDataRegionReplicaSetWithTimeFilter(
+              node.getQualifiedObjectName().getDatabaseName(),
+              deviceEntry.getDeviceID(),
+              node.getTimeFilter());
```

Comment:
```suggestion
          analysis.getDataRegionReplicaSetWithTimeFilter(
              deviceEntry.getDeviceID(),
              node.getTimeFilter());
```
For tree devices, you need to call this method

Reply:
or you should pass the orginal tree dbName instead of tree_view